injector/network_disruption.go

```diff
@@ -135,11 +175,19 @@ func (i networkDisruptionInjector) Inject() error {
 		return fmt.Errorf("unable to exit the given container network namespace: %w", err)
 	}
 
+	go func() {
+		i.config.State.State <- Injected
+	}()
+
 	return nil
 }
 
 // Clean removes all the injected disruption in the given container
-func (i networkDisruptionInjector) Clean() error {
+func (i *networkDisruptionInjector) Clean() error {
+	go func() {
```

Comment:
Should we defer this? We dont want to set this to Cleaned before we're sure the cleaning worked, right?